apollo-router/src/spec/query.rs

```diff
@@ -257,6 +290,58 @@ impl Query {
         query: impl Into<String>,
         schema: &Schema,
         configuration: &Configuration,
+    ) -> Result<Self, SpecError> {
+        Self::parse_with_ast(query, schema, configuration)
+    }
+
+    pub(crate) fn parse_with_hir(
+        query: impl Into<String>,
+        schema: &Schema,
+        configuration: &Configuration,
+    ) -> Result<Self, SpecError> {
+        let query = query.into();
+        let mut compiler = ApolloCompiler::with_recursion_limit(
+            configuration.server.experimental_parser_recursion_limit,
+        );
+        let id = compiler.add_executable(&query, "query");
+        let ast = compiler.db.ast(id);
+
+        // Trace log recursion limit data
+        let recursion_limit = ast.recursion_limit();
+        tracing::trace!(?recursion_limit, "recursion limit data");
```

Comment:
should this be a metric instead?

Reply:
Maybe? It’s copied as-is from the AST-based code, so if it’s to be changed let’s change it in both.

Reply:
Keep the trace level then please :)

Reply:
Hmm it might be interesting
Should we keep the trace level ?